src/database/mongo.js

```diff
@@ -111,38 +112,46 @@
 					if (err) {
 						return callback(err);
 					}
-					createSessionStore();					
+					callback();
 				});
 			} else {
 				winston.warn('You have no mongo password setup!');
-				createSessionStore();
-			}
-
-			function createSessionStore() {
-				var sessionStore;
-				if (nconf.get('redis')) {
-					sessionStore = require('connect-redis')(session);
-					var rdb = require('./redis');
-					rdb.client = rdb.connect();
-
-					module.sessionStore = new sessionStore({
-						client: rdb.client,
-						ttl: 60 * 60 * 24 * 14
-					});
-				} else if (nconf.get('mongo')) {
-					sessionStore = require('connect-mongo')(session);
-					module.sessionStore = new sessionStore({
-						db: db
-					});
-				}
 				callback();
-			}			
+			}
 		});
 	};
 
+	module.initSessionStore = function (callback) {
+		var meta = require('../meta');
+		var sessionStore;
+		if (nconf.get('redis')) {
+			sessionStore = require('connect-redis')(session);
+			var rdb = require('./redis');
+			rdb.client = rdb.connect();
+
+			var ttlDays = 1000 * 60 * 60 * 24 * (parseInt(meta.config.loginDays, 10) || 0);
+			var ttlHours = 1000 * 60 * 60 * (parseInt(meta.config.loginHours, 10) || 0);
+			var ttlMinutes = 1000 * 60 * (parseInt(meta.config.loginMinutes, 10) || 0);
+			var ttlSeconds = 1000 * (parseInt(meta.config.loginSeconds, 10) || 0);
+			var ttlTotal = (ttlDays + ttlHours + ttlMinutes + ttlSeconds) || 1209600000; // Default to 14 days
```

Comment:
Yes, that's correct... and having `loginSeconds` supersede `loginDays` if set. (or hell, even just using `loginSeconds` entirely wouldn't be out of the question. Days is used only because it's user friendly)